packages/react-instantsearch-hooks/src/components/InstantSearchSSRProvider.tsx

```diff
@@ -22,6 +22,14 @@ export function InstantSearchSSRProvider({
   children,
   ...props
 }: InstantSearchSSRProviderProps) {
+  // When <DynamicWidgets> is mounted, a second provider is used above the user-land
+  // <InstantSearchSSRProvider> in `getServerState()`.
+  // To avoid the user's provider overriding the context value with an empty object,
+  // we skip this provider.
+  if (Object.keys(props).length === 0) {
+    return <>{children}</>;
```

Comment:
This should be sufficient?

```suggestion
    return children;
```

Reply:
TS failed without the fragment, as the type became `{} | JSX.Element`, similar issue as to https://github.com/DefinitelyTyped/DefinitelyTyped/issues/33006 I think

Reply:
Yep, I often run into this issue and wrapping with a fragment is the way to go for now.